app/src/main/java/com/example/parkingproject/mvp/model/ReleaseModel.java

```diff
@@ -0,0 +1,57 @@
+package com.example.parkingproject.mvp.model;
+
+import android.util.Log;
+
+import com.example.parkingproject.entities.Parking;
+import com.example.parkingproject.entities.Reservation;
+import com.example.parkingproject.exceptions.ReleaseMoreThanOneMatchException;
+import com.example.parkingproject.exceptions.ReleaseNoMatchException;
+import com.example.parkingproject.exceptions.ReservationListEmptyException;
+
+public class ReleaseModel {
+
+    private final Parking parking;
+
+    public ReleaseModel(Parking parking) {
+        this.parking = parking;
+    }
+
+    public Parking getParking() {
+        return parking;
+    }
+
+    public void parkingRelease(Reservation reservation) throws ReservationListEmptyException, ReleaseMoreThanOneMatchException, ReleaseNoMatchException {
+        if (getParking().getReservationsList().size() <= 0) {
+            throw new ReservationListEmptyException();
+        }
+        if (countSameReservationInList(reservation) > 0) {
+            if (countSameReservationInList(reservation) < 2) {
+                getParking().getReservationsList().remove(reservation);
+            } else {
+                throw new ReleaseMoreThanOneMatchException();
+            }
+        } else {
+            throw new ReleaseNoMatchException();
+        }
+    }
+
+    public int countSameReservationInList(Reservation reservation) {
```

Comment:
Does this need to be public?

Reply:
i used for the test, but now its works without declaring